Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/ExpLog/Order.lean

```diff
@@ -0,0 +1,103 @@
+
+variable {A : Type*} [CStarAlgebra A] [PartialOrder A] [StarOrderedRing A]
+
+open Filter in
+lemma CFC.tendsto_cfc_rpow_sub_one_log {a : A} (ha : IsStrictlyPositive a) :
+    Tendsto (fun p : ℝ => cfc (fun x => p⁻¹ * (x ^ p - 1)) a) (𝓝[>] 0) (𝓝 (CFC.log a)) := by
+  refine tendsto_cfc_fun ?tendsto ?cont
+  case cont =>
+    exact .of_forall fun p ↦ by fun_prop (disch := grind -abstractProof)
+  case tendsto =>
+    let s := {a : A | IsStrictlyPositive a}
+    let f (p : ℝ) (x : ℝ) := if p > 0 then p⁻¹ * (x ^ p - 1) else Real.log x
+    have hmain := tendstoLocallyUniformlyOn_rpow_sub_one_log
+    rw [tendstoLocallyUniformlyOn_iff_forall_isCompact isOpen_Ioi] at hmain
+    exact hmain (spectrum ℝ a) (by grind) (by grind)
+
+open Classical Real in
+/-- `log` is operator monotone. -/
+lemma CFC.log_monotoneOn : MonotoneOn log {a : A | IsStrictlyPositive a} := by
+  /- We have that `log x = lim_{p → 0} p⁻¹ * (x^p - 1)` with uniform convergence on the spectrum of
+  any positive definite operator, which means that `CFC.log a = lim_{p→0} p⁻¹ * (a^p - 1)` by the
+  continuity of the CFC (`tendsto_cfc_fun`). Then, we use the fact that `x^p` is monotone for
+  `p ∈ [0,1]` (`CFC.monotone_nnrpow`) and that the set of monotone functions is closed
+  (`isClosed.monotoneOn`) to conclude the proof. -/
+  let s := {a : A | IsStrictlyPositive a}
+  let f (p : ℝ) := fun a => if a ∈ s then cfc (A := A) (fun x => p⁻¹ * (x ^ p - 1)) a else 0
+  let g := fun a => if a ∈ s then log (A := A) a else 0
+  have hg : s.EqOn g (log (A := A)) := by
+    intro p hp
+    simp [g, hp]
+  refine MonotoneOn.congr ?_ hg
+  refine isClosed_monotoneOn.mem_of_tendsto (f := f) (b := (𝓝[>] 0)) ?tendsto ?eventually
+  case tendsto =>
+    rw [tendsto_pi_nhds]
+    intro a
+    by_cases ha : a ∈ s
+    · have hf : ∀ p, cfc (fun x => p⁻¹ * (x ^ p - 1)) a = f p a := by intro p; simp [f, ha]
+      refine Filter.Tendsto.congr hf ?_
+      have : g a = log a := by simp [g, ha]
+      rw [this]
+      exact tendsto_cfc_rpow_sub_one_log ha
+    · have hg' : g a = 0 := by simp [g, ha]
+      have hf' : ∀ i, f i a = 0 := by simp [f, ha]
+      simp [hg', hf']
+  case eventually =>
+    have h₁ : ∀ᶠ (p : ℝ) in 𝓝[>] 0, p < 1 := by
+      refine Filter.Eventually.filter_mono nhdsWithin_le_nhds ?_
+      exact eventually_lt_nhds (b := 1) (by norm_num)
+    have hp' : ∀ᶠ (p : ℝ) in 𝓝[>] 0, 0 < p := eventually_nhdsWithin_of_forall fun x a => a
+    filter_upwards [h₁, hp'] with p hp hp'
+    have hf : s.EqOn (fun a : A => p⁻¹ • (a ^ p - 1)) (f p) := by
+      intro a ha
+      simp only [ha, ↓reduceIte, f, ← smul_eq_mul]
+      have h₁ : ContinuousOn (fun x => x ^ p) (spectrum ℝ a) := by
+        refine ContinuousOn.rpow_const (by fun_prop) ?_
+        have : IsUnit a := ha.2
+        grind [= spectrum.zero_notMem_iff]
+      have h₂ : ContinuousOn (fun x => x ^ p - 1) (spectrum ℝ a) :=
+        ContinuousOn.sub h₁ (by fun_prop)
+      rw [cfc_smul _ _ _ h₂, cfc_sub _ _ _ h₁ (by fun_prop)]
```

Comment:
`grind` for the win!
```suggestion
      rw [cfc_smul _ (hf := by fun_prop (disch := grind -abstractProof)),
        cfc_sub _ _ (hf := by fun_prop (disch := grind -abstractProof))]
```
I don't quite understand the need for `-abstractProof`. Is this something to do with the way `fun_prop` works? And have we tried switching `cfc_cont_tac` to call `grind` instead of `aesop`?